### PR TITLE
Enable running product-tests with MapR Client

### PR DESCRIPTION
**Pull request by sanjay990**

### CODE REVIEW

**Comment by sanjay990** [comment]
I'll remove the 2nd commit, that's just for my local testing.

File: presto-product-tests/bin/run_on_docker.sh [hunk 1]
```diff
@@ -35,6 +39,18 @@ function stop_unnecessary_hadoop_services() {
...
+function start_mapr_httpfs_service() {
+  HADOOP_MASTER_CONTAINER=$(hadoop_master_container)
+  docker exec ${HADOOP_MASTER_CONTAINER} chown -R mapr /opt/mapr/httpfs
+  docker exec ${HADOOP_MASTER_CONTAINER} maprcli node services -name httpfs -action start -nodes ${HADOOP_HOST_NAME}
+}
+
+function configure_maprclient() {
+  HADOOP_MASTER_CONTAINER=$(hadoop_master_container)
+  PRESTO_MASTER_CONTAINER=$(presto_master_container)
+  docker exec ${PRESTO_MASTER_CONTAINER} /opt/mapr/server/configure.sh -N ${HADOOP_HOST_NAME} -c -C ${HADOOP_HOST_NAME}:7222 -HS ${HADOOP_HOST_NAME}
```
Comment (akshatnair) [inline presto-product-tests/bin/run_on_docker.sh]: Is this port number 7222 configurable in the MapR container? this should be parameterized or atleast a constant.

File: presto-product-tests/conf/docker/common/standard.yml [hunk 2]
```diff
@@ -25,6 +25,9 @@ services:
   presto-master:
     extends:
       service: java-8-base
+    #image: 'teradatalabs/centos6-java8-oracle-mapr:${DOCKER_IMAGES_VERSION}'
+    # Will replace the image once we have that on docker-hub
```
Comment (akshatnair) [inline presto-product-tests/conf/docker/common/standard.yml]: Replace what? I am not following this comment, are you talking about the commented image line above this?

Comment (sanjay990) [reply presto-product-tests/conf/docker/common/standard.yml]: yes, ``image: 'teradatalabs/centos6-java8-oracle-mapr:${DOCKER_IMAGES_VERSION}'``

File: presto-product-tests/bin/run_on_docker.sh (hunk 1 above)
Comment (sanjay990) [reply presto-product-tests/bin/run_on_docker.sh]: https://github.com/Teradata/presto/pull/528/commits/4a3e015040984ee8bf2373d9481c3ef34d9ac78e

File: presto-product-tests/bin/run_on_docker.sh [hunk 3]
```diff
@@ -35,6 +39,18 @@ function stop_unnecessary_hadoop_services() {
   docker exec ${HADOOP_MASTER_CONTAINER} supervisorctl stop zookeeper
 }
 
+function start_mapr_httpfs_service() {
+  HADOOP_MASTER_CONTAINER=$(hadoop_master_container)
+  docker exec ${HADOOP_MASTER_CONTAINER} chown -R mapr /opt/mapr/httpfs
```
Comment (akshatnair) [inline presto-product-tests/bin/run_on_docker.sh]: make /opt/mapr as an env variable for MAPR_HOME  or something and use it everywhere else too.

**Comment by akshatnair** [comment]
LGTM mod one comment.

File: presto-product-tests/bin/run_on_docker.sh [hunk 4]
```diff
@@ -24,6 +24,10 @@ function hadoop_master_container(){
   environment_compose ps -q hadoop-master
 }
 
+function presto_master_container(){
```
Comment (ArturGajowy) [inline presto-product-tests/bin/run_on_docker.sh]: formatting

File: presto-product-tests/bin/run_on_docker.sh [hunk 5]
```diff
@@ -35,6 +39,18 @@ function stop_unnecessary_hadoop_services() {
   docker exec ${HADOOP_MASTER_CONTAINER} supervisorctl stop zookeeper
 }
 
+function start_mapr_httpfs_service() {
+  HADOOP_MASTER_CONTAINER=$(hadoop_master_container)
+  docker exec ${HADOOP_MASTER_CONTAINER} chown -R mapr ${MAPR_HOME}/httpfs
+  docker exec ${HADOOP_MASTER_CONTAINER} maprcli node services -name httpfs -action start -nodes ${HADOOP_HOST_NAME}
```
Comment (ArturGajowy) [inline presto-product-tests/bin/run_on_docker.sh]: why is that done from this script? why doesn't the image do this on startup (in, say, bootstrap.sh?)

File: presto-product-tests/bin/run_on_docker.sh [hunk 6]
```diff
@@ -35,6 +39,18 @@ function stop_unnecessary_hadoop_services() {
   docker exec ${HADOOP_MASTER_CONTAINER} supervisorctl stop zookeeper
 }
 
+function start_mapr_httpfs_service() {
+  HADOOP_MASTER_CONTAINER=$(hadoop_master_container)
+  docker exec ${HADOOP_MASTER_CONTAINER} chown -R mapr ${MAPR_HOME}/httpfs
```
Comment (ArturGajowy) [inline presto-product-tests/bin/run_on_docker.sh]: ditto, this should be done during image startup, or - ideally - during image build

File: presto-product-tests/bin/run_on_docker.sh [hunk 7]
```diff
@@ -35,6 +39,18 @@ function stop_unnecessary_hadoop_services() {
...
+  docker exec ${HADOOP_MASTER_CONTAINER} maprcli node services -name httpfs -action start -nodes ${HADOOP_HOST_NAME}
+}
+
+function configure_maprclient() {
+  HADOOP_MASTER_CONTAINER=$(hadoop_master_container)
```
Comment (ArturGajowy) [inline presto-product-tests/bin/run_on_docker.sh]: this variable is unused

File: presto-product-tests/bin/run_on_docker.sh [hunk 8]
```diff
@@ -202,7 +221,8 @@ HADOOP_LOGS_PID=$!
 # wait until hadoop processes is started
 # retry check_hadoop
 # stop_unnecessary_hadoop_services
-sleep 300
+sleep 150
```
Comment (ArturGajowy) [inline presto-product-tests/bin/run_on_docker.sh]: write a `check_mapr` funciton  or make check_hadoop check for mapr srevice if running against a mapr container.

File: presto-product-tests/conf/docker/common/standard.yml (hunk 2 above)
Comment (ArturGajowy) [reply presto-product-tests/conf/docker/common/standard.yml]: parameterize the image, set the default in compose-commons.sh

**Review by ArturGajowy — CHANGES_REQUESTED** [review]
please avoid commented out code

File: presto-product-tests/bin/run_on_docker.sh (hunk 5 above)
Comment (ArturGajowy) [reply presto-product-tests/bin/run_on_docker.sh]: the `hostname` command could be used to get the value of `${HADOOP_HOST_NAME}` inside the container

Comment (ArturGajowy) [reply presto-product-tests/bin/run_on_docker.sh]:
```
Arturs-MBP:presto arturgajowy$ presto-product-tests/conf/docker/singlenode/compose.sh run hadoop-master hostname
hadoop-master
```

File: presto-product-tests/bin/run_on_docker.sh (hunk 7 above)
Comment (ArturGajowy) [reply presto-product-tests/bin/run_on_docker.sh]: after getting more context from our slack conversation, I see that the `/server/configure.sh` should be run from a specialized `mapr-presto-launcher-wrapper.sh`. The variables used in this function should be defined in that script. Please don't bother with parametrizing the 'HADOOP_HOST_NAME', it's hardwired everywhere anyway. It's ok to just define the variable in that launcher script.

The specialized wrapper should be used by having a `docker/common/mapr.yml` compose file, added to the `mapr` profile. The `image:` override should take place there as well.

File: presto-product-tests/conf/docker/common/standard.yml (hunk 2 above)
Comment (ArturGajowy) [reply presto-product-tests/conf/docker/common/standard.yml]: nvm the 'parameterize the image comment'. This file should not be changed at all for the mapr integration. Instead, there should be a separate `mapr.yml` and a new profile, called `mapr` whose `compose.sh` would include the `mapr.yml` similarly to how all other `compose.sh` work. The image should be overridden in mapr.yml.

File: presto-product-tests/bin/run_on_docker.sh (hunk 4 above)
Comment (ArturGajowy) [reply presto-product-tests/bin/run_on_docker.sh]: This function will be unused after addressing the comments and should be then deleted

File: presto-product-tests/bin/run_on_docker.sh (hunk 5 above)
Comment (ArturGajowy) [reply presto-product-tests/bin/run_on_docker.sh]: please do this in the image's `bootstrap.sh` (modify the docker image accordignly)

**Review by ArturGajowy — CHANGES_REQUESTED** [review]
Updated according to new context from our slack convo

File: presto-product-tests/conf/docker/singlenode-mapr/compose.sh [hunk 9]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+source ${BASH_SOURCE%/*}/../common/compose-commons.sh
+
+docker-compose \
+-f ${BASH_SOURCE%/*}/../common/standard.yml \
+-f ${BASH_SOURCE%/*}/../common/jdbc_db.yml \
+-f ${BASH_SOURCE%/*}/../common/cassandra.yml \
```
Comment (ArturGajowy) [inline presto-product-tests/conf/docker/singlenode-mapr/compose.sh:8]: https://github.com/prestodb/presto/pull/7684 is in master already, but I understand this PR's base does not contain it? I'd prefer we got rid of that before having this in sprint branch

**Review by ArturGajowy — APPROVED** [review]
Looks way simpler and cleaner now :)

File: presto-product-tests/conf/docker/singlenode-mapr/compose.sh (hunk 9 above)
Comment (sanjay990) [reply presto-product-tests/conf/docker/singlenode-mapr/compose.sh:8]: @ArturGajowy : Yes, will remove this before merging with sprint branch. :)

Comment (petroav) [reply presto-product-tests/conf/docker/singlenode-mapr/compose.sh:8]: This isn't going on the sprint branch. It'll stay on the `mapr-support` branch until we figure out what's going to happen.

**Pull request closed without merging**